File: app/src/main/java/korbandras/financeapp/MainActivity.java
package korbandras.financeapp;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.Switch;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatDelegate;

public class MainActivity extends Activity {
    private EditText editTextIncome;
    private EditText editTextExpenses;
    private EditText editTextDueDate;
    private EditText editTextTargetSum;
    private Button calculateButton;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editTextIncome = findViewById(R.id.editTextIncome);
        editTextExpenses = findViewById(R.id.editTextExpenses);
        editTextDueDate = findViewById(R.id.editTextDueDate);
        editTextTargetSum = findViewById(R.id.editTextTargetSum);
        calculateButton = (android.widget.Button) findViewById(R.id.calculateButton);

        calculateButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String income = editTextIncome.getText().toString();
                String expenses = editTextExpenses.getText().toString();
                String dueDate = editTextDueDate.getText().toString();
                String sum = editTextTargetSum.getText().toString();

                Intent intent = new Intent(MainActivity.this, Loading.class);

                intent.putExtra("Income",income);
                intent.putExtra("Expenses",expenses);
                intent.putExtra("DueDate",dueDate);
                intent.putExtra("Sum",sum);

                startActivity(intent);
            }
        });

    }
}

File: app/src/main/java/korbandras/financeapp/MainActivity.kt
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import korbandras.financeapp.R

class MainActivity : AppCompatActivity() {
    private lateinit var editTextIncome: EditText
    private lateinit var editTextExpenses: EditText
    private lateinit var editTextDueDate: EditText
    private lateinit var editTextTargetSum: EditText
    private lateinit var textViewResult: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.newdata)

        editTextIncome = findViewById(R.id.editTextIncome)
        editTextExpenses = findViewById(R.id.editTextExpenses)
        editTextDueDate = findViewById(R.id.editTextDueDate)
        editTextTargetSum = findViewById(R.id.editTextTargetSum)
        textViewResult = findViewById(R.id.textViewResult)
    }

    fun calculateMonthlySavings(view: View) {
        val income = editTextIncome.text.toString().toDoubleOrNull() ?: 0.0
        val expenses = editTextExpenses.text.toString().toDoubleOrNull() ?: 0.0
        val dueDate = editTextDueDate.text.toString().toIntOrNull() ?: 0
        val targetSum = editTextTargetSum.text.toString().toDoubleOrNull() ?: 0.0

        if (income <= expenses || dueDate <= 0 || targetSum <= 0) {
            textViewResult.text = "It's not possible"
        } else {
            val monthlySavings = (targetSum / dueDate) - (expenses - income)
            if (monthlySavings <= 0) {
                textViewResult.text = "It's not possible"
            } else {
                textViewResult.text = "Save $${monthlySavings.format(2)} monthly to reach your target by the due date."
            }
        }
    }

    private fun Double.format(digits: Int) = "%.${digits}f".format(this)
}
